libpkgconf/dependency.c

```diff
@@ -318,6 +318,9 @@ pkgconf_dependency_parse_str(pkgconf_client_t *client, pkgconf_list_t *deplist_h
 
 	memset(cmpname, '\0', sizeof cmpname);
 
+	if (strlen(depends) == 0)
+		return;
+
 	buf_sz = strlen(depends) * 2;
 	buf = calloc(1, buf_sz);
```

Comment:
So this is the original problem. When `buf_sz` is zero, "the behavior is implementation defined (null pointer may be returned, or some non-null pointer may be returned that may not be used to access storage)."
https://en.cppreference.com/w/c/memory/calloc

Reply:
And now we are again at the point that `pkgconf` lacks running tests on Windows/MinGW.

Reply:
FTR running tests on Windows is a subissue of #406.